src/ElasticLoadBalancingV2/ElasticLoadBalancingV2Client.php

```diff
@@ -64,4 +67,38 @@
  * @method \Aws\Result setSubnets(array $args = [])
  * @method \GuzzleHttp\Promise\Promise setSubnetsAsync(array $args = [])
  */
-class ElasticLoadBalancingV2Client extends AwsClient {}
+class ElasticLoadBalancingV2Client extends AwsClient {
+
+    public function __construct(array $args)
+    {
+        if (!isset($args['signing_name'])) {
+            $args['signing_name'] = 'elasticloadbalancing';
+        }
+
+        parent::__construct($args);
+        $stack = $this->getHandlerList();
+        $stack->appendBuild($this->getEndpointMiddleware(), 'elbv2.elb_endpoint');
+    }
+
+    /**
+     * Provide a middleware that resolves ElbV2 endpoints with Elb endpoints
+     *
+     * @return \Closure
+     */
+    private function getEndpointMiddleware()
+    {
+        $region = $this->getRegion();
+        return static function (callable $handler) use ($region) {
+            return function (
+                CommandInterface $command,
+                RequestInterface $request = null
+            ) use ($handler, $region) {
+                $request = $request->withUri(
+                    $request->getUri()
+                        ->withHost("elasticloadbalancing.{$region}.amazonaws.com")
```

Comment:
I think this should be configured in the constructor, not each time a request is created. This just overwrites the host header, so if you wanted to hit a custom endpoints (like a test endpoint for example), this would not allow that.